ste/mgr-JobMgr.go

```diff
@@ -188,10 +192,92 @@ type jobMgr struct {
 
 	// only a single instance of the prompter is needed for all transfers
 	overwritePrompter *overwritePrompter
+	// only a single instance of the user delegation authentication manager is needed for all transfers
+	udam             *userDelegationAuthenticationManager
+	udamCreationLock sync.Mutex // This is a sanity check because it *may* be possible to have AddJobPart called multiple times at once if transfers get scheduled fast enough.
 }
 
 ////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
 
+func (jm *jobMgr) GetUserDelegationAuthenticationManagerInstance() *userDelegationAuthenticationManager {
+	return jm.udam
+}
+
+func (jm *jobMgr) setupUserDelegationAuthManager(src string) {
+	// Check if it doesn't already exist.
+	if jm.udam == nil {
+		// lock up to ensure nobody steps over us.
+		jm.udamCreationLock.Lock()
+		defer jm.udamCreationLock.Unlock()
+
+		// If somebody else has already done it by the time we obtained the lock, we're good to go.
+		if jm.udam != nil {
+			return
+		}
+
+		// If the OAuth token is empty, this will fail anyway.
+		// Set an empty UDAM instance that does nothing.
+		if jm.getInMemoryTransitJobState().credentialInfo.OAuthTokenInfo == (common.OAuthTokenInfo{}) {
+			jm.udam = &userDelegationAuthenticationManager{}
+			return
+		}
+
+		srcURL, err := url.Parse(src)
+
+		// no reason for this to fail, ever.
+		common.PanicIfErr(err)
+
+		// get a service-level URL
+		srcBlobURLParts := azblob.NewBlobURLParts(*srcURL)
+		srcBlobURLParts.BlobName = ""
+		srcBlobURLParts.ContainerName = ""
+
+		// create credentials, xfer options
+		credOption := common.CredentialOpOptions{
```

Comment:
TODO: consider this logging strategy

It only ends up outputting the actual attempted request itself. This works to show success or failure at the creation of a UDK, but it's something that is only understood by the developer at this point.